File: backend/app/schemas/deals.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class DealView(BaseModel):
    retailer_name: str
    retailer_code: str
    store_id: int
    store_location: str
    store_number: str | None = None
    product_title: str
    brand: str | None = None
    sku_item_id: str
    upc_gtin: str | None = None
    current_price: Decimal
    original_price: Decimal | None = None
    discount_percent: float = Field(default=0.0)
    clearance_label: str | None = None
    stock_level: int | None = None
    deal_confidence: float = Field(default=0.0)
    product_url: str | None = None
    image_url: str | None = None
    estimated_resale_profit: Decimal
    estimated_roi: Decimal
    updated_at: datetime


class DealQuery(BaseModel):
    retailer_codes: list[str] = []
    zip_code: str | None = "78701"
    radius_miles: int = 25
    store_ids: list[int] = []
    compare_store_ids: list[int] = []
    min_profit: float | None = None
    min_roi: float | None = None
    search: str | None = None
    clearance_only: bool = False
    high_resale_only: bool = False
    sort_by: str = "highest_roi"
    page: int = 1
    page_size: int = 25


class StoreView(BaseModel):
    id: int
    retailer_code: str
    retailer_name: str
    store_number: str
    name: str
    city: str
    state: str
    zip_code: str
    distance_miles: float

File: frontend/src/types/deals.ts
export type RetailerCode = 'homedepot' | 'costco' | 'walmart' | 'target';

export interface DealRow {
  retailer_name: string;
  retailer_code: RetailerCode;
  store_id: number;
  store_location: string;
  store_number?: string;
  product_title: string;
  brand?: string;
  sku_item_id: string;
  upc_gtin?: string;
  current_price: number;
  original_price?: number;
  discount_percent: number;
  clearance_label?: string;
  stock_level?: number;
  deal_confidence: number;
  product_url: string;
  image_url: string;
  estimated_resale_profit: number;
  estimated_roi: number;
  updated_at: string;
}

export interface StoreOption {
  id: number;
  retailer_code: RetailerCode;
  retailer_name: string;
  store_number: string;
  name: string;
  city: string;
  state: string;
  zip_code: string;
  distance_miles: number;
}
